sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/optimizer/Optimizer.scala

```diff
@@ -958,6 +958,26 @@ object PushDownPredicate extends Rule[LogicalPlan] with PredicateHelper {
 
       project.copy(child = Filter(replaceAlias(condition, aliasMap), grandChild))
 
+    // Push [[Filter]] operators through [[Window]] operators. Parts of the predicate that can be
+    // pushed beneath must satisfy three conditions:
+    // 1. involving one and only one column that is part of window partitioning key.
+    // 2. Window partitioning key should be just a sequence of [[AttributeReference]].
```

Comment:
Yeah, thank you very much! I just added an extra condition to ensure that Analyzer converts all the compound expressions to alias. Added a couple of test cases to ensure it. 

To enable predicate push down when the partitioning columns is `a + b` and the predicate is `a + b > 3`, we need to add a rule in Analyzer for converting the expressions to the underlying alias that has the exactly same original expressions. This will be submitted in a separate PR.